catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -131,215 +112,110 @@ public UpdateResponse version(UpdateResponse updateResponse)
-                            versionedMetacards,
-                            props)));
-            return Optional.ofNullable(createResponse)
-                    .map(CreateResponse::getProcessingErrors)
-                    .map(Set::isEmpty)
-                    .orElse(false);
-        });
-
-        return Optional.of(transactionKey);
+        setResourceUriForContent(/*mutable*/ versionMetacards, createStorageResponse);
+
+        CreateResponse createResponse = storeVersionMetacards(versionMetacards);
+
+        return updateStorageResponse;
     }
 
     /**
      * Versions deleted {@link Metacard}s.
      *
      * @param deleteResponse Versions this responses deleted metacards
      */
-    public void version(DeleteResponse deleteResponse) {
+    public DeleteResponse version(DeleteResponse deleteResponse)
+            throws SourceUnavailableException, IngestException {
         if (doSkip(deleteResponse)) {
-            return;
+            return deleteResponse;
         }
         setSkipFlag(deleteResponse);
 
-        List<Metacard> versionedMetacards = deleteResponse.getDeletedMetacards()
+        Map<String, List<ContentItem>> contentItems = getContentItems(deleteResponse);
+        Action action = contentItems.isEmpty() ? Action.DELETED : Action.DELETED_CONTENT;
+
+        Map<String, Metacard> versionedMap =
+                getVersionMetacards(deleteResponse.getDeletedMetacards(),
+                        action,
+                        (Subject) deleteResponse.getRequest()
+                                .getProperties()
+                                .get(SecurityConstants.SECURITY_SUBJECT));
+
+        CreateStorageResponse createStorageResponse = versionContentItems(contentItems,
+                versionedMap);
+        if (createStorageResponse != null) {
+            setResourceUriForContent(/*Mutable*/ versionedMap, createStorageResponse);
+        }
+
+        CreateResponse createResponse =
+                executeAsSystem(() -> catalogProvider().create(new CreateRequestImpl(new ArrayList<>(
+                        versionedMap.values()))));
+        String emailAddress = SubjectUtils.getEmailAddress((Subject) deleteResponse.getProperties()
+                .get(SecurityConstants.SECURITY_SUBJECT));
+        List<Metacard> deletedMetacards = versionedMap.entrySet()
                 .stream()
-                .map(mc -> new MetacardVersion(mc,
-                        MetacardVersion.Action.DELETED,
-                        getSubject.get()))
+                .map(s -> new DeletedMetacardImpl(s.getKey(),
+                        emailAddress,
+                        s.getValue()
+                                .getId(),
+                        MetacardVersionImpl.toMetacard(s.getValue(), metacardTypes)))
                 .collect(Collectors.toList());
 
-        CreateResponse createResponse =
+        CreateResponse deleteTrackResponse =
```

Comment:
Not important, but you're never using this object either; you only care about the creation of it.